### PR TITLE
 [ADD] module 'mrp_bom_structure_xls'

### PR DESCRIPTION
**Pull request by JordiBForgeFlow**
This module was written to extend the MRP capabilities of Odoo, and
allows users to export the BOM structure to MS Excel .XLS format.

### CODE REVIEW

**Comment by JordiBForgeFlow** [comment]
Travis fails due to depending library "report_xls" is not in this repo. Not sure how to proceed in this case..

**Comment by tafaRU** [comment]
@jbeficent see https://github.com/Eficent/manufacture-reporting/pull/1, it should fix the error in travis build. By the way, thanks a lot for your contribution!

**Comment by JordiBForgeFlow** [comment]
Thanks @tafaRU How to re-run Travis?

**Comment by tafaRU** [comment]
@jbeficent I think it's enough to merge my PR.

**Comment by JordiBForgeFlow** [comment]
@tafaRU Not sure why runbot would fail on this one with message "ImportError: No module named report_xls" if the oca-dependencies have been added. Any clues?

**Comment by tafaRU** [comment]
@jbeficent oca-dependecies fixed travis build, runbot build error I don't think is related.
I suppose the problem should be fixed adding the OCA/reporting-engine repo to the manufacture-reporting runbot. Please send a mail to contribute@odoo-community.org  or ask to @gurneyalex on #oca irc channel.

**Comment by gurneyalex** [comment]
@jbeficent I updated the runbot config, and restarted a build. 

thanks for the notice.

**Comment by JordiBForgeFlow** [comment]
Thanks @gurneyalex ! However I see that in the PR runbot is red, not yellow. Is that ok?

**Comment by gurneyalex** [comment]
@jbeficent can't seem to restart. Can you run `git commit --amend` and `git push -f` to force a rebuild?

**Comment by JordiBForgeFlow** [comment]
@gurneyalex Done. But sems that runbot and Travis are not starting.

**Comment by JordiBForgeFlow** [comment]
@gurneyalex Thanks. It's now green. Coveralls is red, but not sure what is it's role.

**Comment by gurneyalex** [comment]
@jbeficent Coveralls role is red because you should add a simple automated test to the module exercising the code by generating a report. You'll get bonus points in my view if you also check the contents of the report. But just generating the report will ensure the code runs and improve the test coverage of the project.

**Comment by pedrobaeza** [comment]
Any news on this?

**Comment by gurneyalex** [comment]
:+1:

**Comment by JordiBForgeFlow** [comment]
This one is ready to be merged. Anyone can quickly check and approve?

File: mrp_bom_structure_xls/README.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+Export BOM Structure to Excel
```
Comment (pedrobaeza) [inline mrp_bom_structure_xls/README.rst:6]: Please update template to the newest one.

File: mrp_bom_structure_xls/__init__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline mrp_bom_structure_xls/__init__.py]: Short headers

File: mrp_bom_structure_xls/__openerp__.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+{
+    'name': "Export BOM Structure to Excel",
+    'version': '1.0',
```
Comment (pedrobaeza) [inline mrp_bom_structure_xls/__openerp__.py]: 8.0.1.0.0

File: mrp_bom_structure_xls/__openerp__.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    'category': 'Manufacturing',
+    'author': "Eficent",
```
Comment (pedrobaeza) [inline mrp_bom_structure_xls/__openerp__.py]: Add OCA

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: mrp_bom_structure_xls/__openerp__.py (hunk 4 above)
Comment (LoisRForgeFlow) [reply mrp_bom_structure_xls/__openerp__.py]: Ok, We will attend all this.

**Comment by LoisRForgeFlow** [comment]
why are there no running test (travis, runbot...) for this PR?

**Comment by JordiBForgeFlow** [comment]
Because it is an export to XLS. Not sure how it would be possible to.add

**Comment by AaronHForgeFlow** [comment]
coveralls is not referring any file of this PR

**Comment by JordiBForgeFlow** [comment]
@gurneyalex @moylop260  Seems that runbot is not running on this repo. Do you know why?

**Comment by moylop260** [comment]
@jbeficent 
The issue is https://github.com/OCA/runbot-addons/issues/115

**Comment by JordiBForgeFlow** [comment]
Thanks @moylop260. Created PR https://github.com/OCA/manufacture-reporting/pull/15

**Comment by gurneyalex** [comment]
PR #15 merged. You need to rebase your PRs to get runbot builds.

File: mrp_bom_structure_xls/README.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+   :alt: License: AGPL-3
+
+=============================
+Export BOM Structure to Excel
+=============================
+
+This module was written to extend the MRP capabilities of Odoo, and
+allows users to export the BOM structure to MS Excel .XLS format.
+
+Installation
```
Comment (JordiBForgeFlow) [inline mrp_bom_structure_xls/README.rst]: Remove this section, as it is not used

File: mrp_bom_structure_xls/README.rst [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+============
+
+No additional installation instructions are required.
+
+Configuration
```
Comment (JordiBForgeFlow) [inline mrp_bom_structure_xls/README.rst]: Remove this section, as it is not used

File: mrp_bom_structure_xls/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    'version': '8.0.1.0.0',
+    'category': 'Manufacturing',
+    'author': "Eficent, Odoo Community Association (OCA)",
+    'website': 'https://github.com/OCA/manufacture-reporting',
+    'license': 'AGPL-3',
+    "depends": ['report_xls', 'mrp'],
+    "data": [
+        'report/bom_structure_xls.xml',
+    ],
+    'test': [],
```
Comment (tafaRU) [inline mrp_bom_structure_xls/__openerp__.py]: Please remove empty keys

**Comment by JordiBForgeFlow** [comment]
@tafaRU @pedrobaeza This one is ready to merge.

**Review by AaronHForgeFlow — APPROVED** [review]
Tested on runbot :+1:

**Review by tafaRU — APPROVED** [review]
Thanks!

**Review by moylop260 — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
Tested :+1:

**Pull request merged**